components/renku_data_services/notebooks/api/amalthea_patches/init_containers.py

```diff
@@ -79,21 +80,170 @@ def git_clone(server: "UserServer") -> list[dict[str, Any]]:
+    if not repositories:
+        return None
+
+    etc_cert_volume_mount = get_certificates_volume_mounts(
+        server.config,
+        custom_certs=False,
+        etc_certs=True,
+        read_only_etc_certs=True,
+    )
+
+    user_is_anonymous = not server.user.is_authenticated
+    prefix = "GIT_CLONE_"
+    env = [
+        {
+            "name": f"{prefix}WORKSPACE_MOUNT_PATH",
+            "value": server.workspace_mount_path.absolute().as_posix(),
+        },
+        {
+            "name": f"{prefix}MOUNT_PATH",
+            "value": server.work_dir.absolute().as_posix(),
+        },
+        {
+            "name": f"{prefix}LFS_AUTO_FETCH",
+            "value": "1" if server.server_options.lfs_auto_fetch else "0",
+        },
+        {
+            "name": f"{prefix}USER__USERNAME",
+            "value": server.user.email,
+        },
+        {
+            "name": f"{prefix}USER__RENKU_TOKEN",
+            "value": str(server.user.access_token),
+        },
+        {"name": f"{prefix}IS_GIT_PROXY_ENABLED", "value": "0" if user_is_anonymous else "1"},
+        {
+            "name": f"{prefix}SENTRY__ENABLED",
+            "value": str(server.config.sessions.git_clone.sentry.enabled).lower(),
+        },
+        {
+            "name": f"{prefix}SENTRY__DSN",
+            "value": server.config.sessions.git_clone.sentry.dsn,
+        },
+        {
+            "name": f"{prefix}SENTRY__ENVIRONMENT",
+            "value": server.config.sessions.git_clone.sentry.env,
+        },
+        {
+            "name": f"{prefix}SENTRY__SAMPLE_RATE",
+            "value": str(server.config.sessions.git_clone.sentry.sample_rate),
+        },
+        {"name": "SENTRY_RELEASE", "value": os.environ.get("SENTRY_RELEASE")},
+        {
+            "name": "REQUESTS_CA_BUNDLE",
+            "value": str(Path(etc_cert_volume_mount[0]["mountPath"]) / "ca-certificates.crt"),
+        },
+        {
+            "name": "SSL_CERT_FILE",
+            "value": str(Path(etc_cert_volume_mount[0]["mountPath"]) / "ca-certificates.crt"),
+        },
+    ]
+    if server.user.is_authenticated:
+        if server.user.email:
+            env.append(
+                {"name": f"{prefix}USER__EMAIL", "value": server.user.email},
+            )
+        if server.user.full_name:
+            env.append(
+                {
+                    "name": f"{prefix}USER__FULL_NAME",
+                    "value": server.user.full_name,
+                },
+            )
+        elif server.user.first_name or server.user.last_name:
+            full_name = server.user.first_name or ""
+            if full_name == "":
+                full_name = server.user.last_name or ""
+            else:
+                full_name += " " + (server.user.last_name or "")
+            env.append(
+                {
+                    "name": f"{prefix}USER__FULL_NAME",
+                    "value": server.user.full_name,
+                },
+            )
```

Comment:
nitpic: same here, if this was a method on user, it'd simplify things a bit

Reply:
done